### DISCUSSION TITLE
Can't open URL with multiple query params

### DISCUSSION THREAD

**Discussion by modojodo** [johnlindquist/kit#782] [General]
Hi there, I have a script use case where I want to open a custom URL that includes the query params. If I use he `browse()` function, the browser ignores anything after the `&`. `focusTab()` is the only function that considers the query params after `&` but that doesn't suit my use case as I have to open multiple tabs in an iteration and it seems to only open the tab for the last iteration.

I have tried using `encodeURI()` as well to create a URL for `browse()` and `run()` functions but that does work either.

Example: 

If the URL is  `www.foo.com/?abc=123&bar=foo`

Then the browse function or run function would open `www.foo.com/?abc=123` and ignores the rest.

**Comment by johnlindquist** [comment]
Try:

```js
exec(`open 'https://www.foo.com/?abc=123&bar=foo'`)
```

The `&` has special meaning in shells, so we need to wrap in quotes to ensure it's interpreted as a string.

I'll update `browse` so that it always wraps the URL you pass in in quotes. Currently, browse is literally:

```js
global.browse = async (path: string) => {
  return global.exec(`open ${path}`)
}
```

**Reply by modojodo** [reply]
Thank you so much that worked.

It would be much appreciated if you can update the browse function! Thanks again! 🙌
